### PR TITLE
Support storing node/edge feature as float16.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/construct_graph.py [hunk 1]
```diff
@@ -39,7 +39,11 @@
 from .id_map import NoopMap, IdMap, map_node_ids
 from .utils import multiprocessing_data_read, ExtMemArrayMerger, partition_graph
 
-def parse_node_data(in_file, feat_ops, label_ops, node_id_col, read_file):
+FEATURE_STORE_PRECISION_FP32 = "fp32"
+FEATURE_STORE_PRECISION_FP16 = "fp16"
+
+def parse_node_data(in_file, feat_ops, label_ops, node_id_col,
+                    read_file, feat_dtype=np.float32):
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/construct_graph.py:46]: i would prefer that you create a feature operator that converts the data type. in this way, we don't need to pass anything argument to any of the functions.

Comment (zheng-da) [reply python/graphstorm/gconstruct/construct_graph.py:46]: in this way, users can define output data type in the JSON file to define the output data type for each feature.

File: python/graphstorm/gconstruct/file_io.py [hunk 2]
```diff
@@ -244,13 +245,27 @@ def __getitem__(self, idx):
...
         """ Return Numpy array.
         """
-        return self._arr[:]
+        res = self._arr[:]
+        if self._out_dtype is not None:
+            res = res.astype(self._out_dtype)
+        return res
+
+    def astype(self, dtype):
+        """ Set the output dtype.
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/file_io.py:265]: specify that `dtype` is numpy dtype.

**Comment by zheng-da** [comment]
Can you update the README file to show out_dtype is supported.

**Comment by zheng-da** [comment]
originally, I was thinking of another Transform that convert the data type if `out_dtype` is specified in a feature block. But I guess this is also fine. The only problem is that all Transform operation now needs to support an out_dtype.

**Comment by classicsong** [comment]
> Can you update the README file to show out_dtype is supported.

Sure

**Comment by classicsong** [comment]
> originally, I was thinking of adding another Transform that converts the data type if `out_dtype` is specified in a feature block. But I guess this is also fine. The only problem is that all Transform operation now needs to support an out_dtype.

It it not mandatory for all Transform to support out_dtype. For example, Tokenizer ignores out_dtype.

**Review by zheng-da — APPROVED** [review]

**Pull request merged**